src/mbgl/actor/mailbox.cpp

```diff
@@ -88,19 +90,28 @@ void Mailbox::push(std::unique_ptr<Message> message) {
         }
     }};
 
+    MLN_TRACE_ZONE(push lock)
     std::lock_guard<std::mutex> pushingLock(pushingMutex);
 
     if (closed) {
         state = State::Abandoned;
         return;
     }
 
-    std::lock_guard<std::mutex> queueLock(queueMutex);
-    bool wasEmpty = queue.empty();
-    queue.push(std::move(message));
-    auto guard = weakScheduler.lock();
-    if (wasEmpty && weakScheduler) {
-        weakScheduler->schedule(schedulerTag, makeClosure(shared_from_this()));
+    bool wasEmpty;
```

Comment:
Should be initialized to something.

Reply:
Even though it's unconditionally set?  That will generate "value is never used" warnings on some compilers.

Reply:
C++ Code Guidelines say yes. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es20-always-initialize-an-object